### PR TITLE
Feat/user agent enhancements/part 1 cognito

### PR DESCRIPTION
**Pull request by erinleigh90**
#### Description of changes
User Agent Enhancements Part I for amazon-cognito-identity-js:
- Adds type for CustomUserAgent and enums for Category, CategoryAction, and Framework
- Updates core Platform to build the user agent according to the new format `aws-amplify/5.0.14 (1:1,2:2,3:1,4:0)`
- NOTE: Since amazon-cognito-identity-js should remain independent of aws-amplify, there are some duplicates of files/content from the feat/user-agent-enhancements/part1-core branch

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cshfang — COMMENTED** [review]
Kind of have the same general comments as #11122 since this is sort of just copying that

File: packages/amazon-cognito-identity-js/src/constants.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export const Category = {
+	Auth: '1:1',
```
Comment (stocaaro) [inline packages/amazon-cognito-identity-js/src/constants.js:2]: I've lost the thread a bit on this. I think these are going to be translated to a different format, right?  Will that happen in a subsequent change or before this merges?

**Review by stocaaro — COMMENTED** [review]
Change seems pretty straight forward. Have one question below. What's keeping this change in draft?

File: packages/amazon-cognito-identity-js/src/UserAgent.js [hunk 2]
```diff
@@ -23,4 +25,15 @@ export const appendToCognitoUserAgent = content => {
 };
 
 // class for defining the amzn user-agent
+export const getAmplifyUserAgent = customUserAgent => {
```
Comment (AllanZhengYP) [inline packages/amazon-cognito-identity-js/src/UserAgent.js:28]: Use `getAmplifyUserAgentString` just to be consistent with the other PR?

**Pull request closed without merging**